packages/browser-wallet/src/popup/store/transactions.ts

```diff
@@ -43,28 +44,51 @@ const monitorTransactionStatus = (genesisHash: string) => {
 
 const pendingTransactionsAtom = (() => {
     const base = atomWithChromeStorage<string[]>(ChromeStorageKey.PendingTransactions, []);
-
-    return atom<BrowserWalletAccountTransaction[], BrowserWalletAccountTransaction[]>(
+    const parsed = atom<BrowserWalletAccountTransaction[], BrowserWalletAccountTransaction[]>(
         (get) => {
-            const pendingJson = get(base);
-            const pending: BrowserWalletAccountTransaction[] = pendingJson.map(parse);
-            const network = get(networkConfigurationAtom);
+            const pending: BrowserWalletAccountTransaction[] = get(base).map(parse);
+            return pending;
+        },
+        (_, set, update) => {
+            set(base, update.map(stringify));
+        }
+    );
+
+    const derived = atom<BrowserWalletAccountTransaction[], BrowserWalletAccountTransaction[]>(
+        (get) => get(parsed),
+        (get, set, update) => {
+            const pending = [...get(parsed), ...update];
+            set(parsed, pending);
 
+            const network = get(networkConfigurationAtom);
             const monitor = monitorTransactionStatus(network.genesisHash);
+
             pending.forEach(async ({ transactionHash }) => {
                 const shouldRemove = await monitor(transactionHash);
+                const currentNetwork = get(networkConfigurationAtom);
 
-                if (shouldRemove) {
-                    sessionPendingTransactions.set(pendingJson.filter((json) => !json.includes(transactionHash)));
+                if (!shouldRemove) {
+                    return;
                 }
-            });
 
-            return pending;
-        },
-        (_, set, update) => {
-            set(base, update.map(stringify));
+                const networkChanged = network.genesisHash !== currentNetwork.genesisHash;
+                const nextPending = pending.filter((p) => p.transactionHash !== transactionHash);
+
+                if (!networkChanged) {
+                    set(parsed, nextPending);
+                } else {
+                    const spt = useIndexedStorage(sessionPendingTransactions, async () => currentNetwork.genesisHash);
+                    spt.set(nextPending.map(stringify));
+                }
+            });
         }
     );
+
+    derived.onMount = (startMonitoring) => {
+        startMonitoring([]);
+    };
```

Comment:
What does this do, if anything?

Reply:
Well... it starts monitoring? 😄 

When the atom mounts (i.e. is used in a view), we check all the pending transactions stored in session. Imagine the use case where you

1. Submit a transaction
2. Close the wallet
3. Re-open the wallet

Here, we have to re-initiate the monitoring, as it stops when the user closes the wallet.

Reply:
Added a comment that explains the behaviour